### PR TITLE
output error msg on Serial if HV charging fails

### PR DESCRIPTION
**Pull request by ThomasWaldmann**
also:
- add a constant MAX_CHARGE_PULSES
- lowered MAX_CHARGE_PULSES from 1000 to 333 (so it is << 1s as expected by main loop recharge)
- cosmetic changes (3rd commit)

### CODE REVIEW

File: multigeiger/multigeiger.ino [hunk 1]
```diff
@@ -137,6 +137,9 @@ enum {SEND_CPM,SEND_BME};
 // Measurement interval (default 2.5min) [sec]
 #define MEASUREMENT_INTERVAL 150
 
+// How many HV capacitor charge pulses to generate before giving up:
+#define MAX_CHARGEPULSES 1000
```
Comment (ThomasWaldmann) [inline multigeiger/multigeiger.ino]: btw, i noticed that with the current value of 1000, charging will take up to ~2.6s which is way too long considering we recharge in the main loop every 1000ms anyway, see:
```
  // Pulse the high voltage if we got enough GMC pulses to update the display or at least every 1000ms.
```

File: multigeiger/multigeiger.ino [hunk 2]
```diff
@@ -681,8 +685,10 @@ int jb_HV_gen_charge__chargepulses() {
     digitalWrite(PIN_HV_FET_OUTPUT, LOW);               // turn the HV FET off
     delayMicroseconds(1000);
     chargepulses++;
-  } while ((chargepulses < 1000) && !isr_GMC_cap_full); // either a timeout or a capacitor full interrupt stops this loop
+  } while ((chargepulses < max_chargepulses) && !isr_GMC_cap_full); // either a timeout or a capacitor full interrupt stops this loop
   time2hvpulse = millis();                              // we just pulsed, so restart timer
+  if((chargepulses == max_chargepulses) && !isr_GMC_cap_full)
+    Serial.println("Error: HV charging failed!");       // pulsed a lot, but still the capacitor is not at desired voltage
```
Comment (ThomasWaldmann) [inline multigeiger/multigeiger.ino]: just a quickfix here to not let the error be completely silent.

for users not having connected the serial terminal it of course would be better to also update some global "hv_good" state and display it on the oled display.

**Comment by ThomasWaldmann** [comment]
Usually I see that it needs 1 pulse (from serial output). So lowering from 1000 to 333 should be no problem at all (and maybe could be lowered again to a way smaller value later).

**Comment by ThomasWaldmann** [comment]
@WwwBoehriDe What do you think?

It's a new concept to me that functions have their creators initials in their name. :-)

**Comment by WwwBoehriDe** [comment]
Hi,
for me it is no problem to change the name of the Function. It was just for me to separate between Arduino- and Selfmade functions. No worries!

The Problem I see is that the limit covers two cases:
- rechargeing that happens every second. For this the limit is OK.
- but it also is the limit for the first charging. When the Main-Capacitor is charged 0=>400 V. Here the pulses could vary (I did not further investigat) a lot over PVT. Probably here a Limit of 333 could be too low. And here we don't want to see the Warning in the Log. 
Probably its a good sollution to suppress warnings the first 5 charge-runs.

**Comment by ThomasWaldmann** [comment]
i can supply more charge pulses for first call from setup() as the max pulses is a func param now.



On January 12, 2020 12:53:02 PM GMT+01:00, WwwBoehriDe <notifications@github.com> wrote:
>Hi,
>for me it is no problem to change the name of the Function. It was just
>for me to separate between Arduino- and Selfmade functions. No worries!
>
>The Problem I see is that the limit covers two cases:
>- rechargeing that happens every second. For this the limit is OK.
>- but it also is the limit for the first charging. When the
>Main-Capacitor is charged 0=>400 V. Here the pulses could vary (I did
>not further investigat) a lot over PVT. Probably here a Limit of 333
>could be too low. And here we don't want to see the Warning in the Log.
>
>Probably its a good sollution to suppress warnings the first 5
>charge-runs.
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/ecocurious/MultiGeiger/pull/131#issuecomment-573407549

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by ThomasWaldmann** [comment]
Merging now "as is", later more pulses from setup() call.

**Pull request merged**